lib/common/grpc-service.js

```diff
@@ -325,7 +325,8 @@ GrpcService.prototype.requestStream = function(protoOpts, reqOpts) {
 
     request: function() {
       return service[protoOpts.method](reqOpts, grpcOpts)
-        .on('status', function(status) {
+        .on('metadata', function() {
+          var status = this.received_status || { code: 0 };
```

Comment:
You should listen for the `status` event from the grpc call to get the grpc status. But that event is emitted when the call completes, so you shouldn't be expecting it at the beginning of a call.